src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -0,0 +1,569 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataMultipartMixedBatchWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.MultipartMixed
+{
+    #region Namespaces
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+#if PORTABLELIB
+    using System.Threading.Tasks;
+#endif
+    #endregion Namespaces
+
+    internal sealed class ODataMultipartMixedBatchWriter : ODataBatchWriter

```

Comment:
>internal [](start = 4, length = 8)

why was the ODataBatchWriter public but this derived type is internal?

Reply:
This is by design. The ODataBatchWriter is a public interface/type so that it must be public.

ODataMultipartMixedBatchWriter should always be instantiated from factory method such as ODataRawOutputContext.CreateBatchWriterImplementation(), and never be instantiated directly from external code.